components/grade-table.js

```diff
@@ -1,31 +1,57 @@
 class GradeTable {
-  constructor(tableElement){
+  constructor(tableElement, noGradesElement){
     this.tableElement = tableElement;
+    this.noGradesElement = noGradesElement;
   }
   updateGrades(grades){
     var tBody = this.tableElement.querySelector('tbody');
-    var tBody2 = tBody.children;
-    var array = [];
-    for(var i=0; i < tBody2.length; i++){
-      array.push(tBody2[i])
+    tBody.innerHTML = "";
+    for (var i = 0; i < grades.length; i++) {
+      var studentGrades = this.renderGradeRow(grades[i], this.deleteGrade);
+      tBody.append(studentGrades);
     }
-    for (var i=0; i< array.length;i++){
-      array[i].remove();
-    }
-    for(var i =0; i<grades.length;i++){
-      var row = document.createElement('tr');
-
-      var name = document.createElement('td');
-      name.textContent = grades[i].name;
-      row.appendChild(name);
-      var course = document.createElement('td');
-      course.textContent = grades[i].course;
-      row.appendChild(course);
-      var grade = document.createElement('td');
-      grade.textContent = grades[i].grade;
-      row.appendChild(grade);
-      tBody.appendChild(row);
+
+
+    var p = document.querySelector('p');
```

Comment:
Please use the reference you saved in the `constructor`, rather than re-querying the whole `document`.